sycl/include/sycl/accessor.hpp

```diff
@@ -1793,6 +1794,51 @@ class __SYCL_SPECIAL_CLASS accessor :
     return constant_ptr<DataT>(getPointerAdjusted());
   }
 
+  // accessor::has_property for runtime properties is only available in host
+  // code. This restriction is not listed in the core spec and will be added in
+  // future versions.
+  template <typename Property>
+  typename sycl::detail::enable_if_t<
+      !ext::oneapi::is_compile_time_property<Property>::value, bool>
+  has_property() const noexcept {
+#ifndef __SYCL_DEVICE_ONLY__
+    return AccessorBaseHost::impl->MPropertyList.has_property<Property>();
+#else
+    return false;
+#endif
+  }
+
+  // accessor::get_property for runtime properties is only available in host
+  // code. This restriction is not listed in the core spec and will be added in
+  // future versions.
+  template <typename Property,
+            typename = typename sycl::detail::enable_if_t<
+                !ext::oneapi::is_compile_time_property<Property>::value>>
+  Property get_property() const {
+#ifndef __SYCL_DEVICE_ONLY__
+    return AccessorBaseHost::impl->MPropertyList.get_property<Property>();
+#else
+    throw sycl::invalid_object_error("The property is not found",
+                                     PI_ERROR_INVALID_VALUE);
```

Comment:
Do I read it correctly that you are trying to throw an exception from device code here? This doesn't look correctly: calling this method in a kernel will likely result in an obscure compilation error, which could even originate from the SPIR-V translator.

I would suggest that we insert some other kind of `noop` here

Reply:
Okay, as discussed let's just return a default property constructor [6d19248](https://github.com/intel/llvm/pull/6614/commits/6d1924846ce72d04b390a4fcc3ecfcd028274eef)